File: 224-basic-calculator/basic-calculator.ts
function calculate(s: string): number {
    const stack: number[] = []
    let currentNumber: number = 0
    let result: number = 0
    let sign: number = 1
    console.log(isNaN(Number(' ')))

    for (let i = 0; i < s.length; i++) {
        const char = s[i]
        console.log('number = ' + char)
        console.log(!isNaN(Number(char)))
        if (!isNaN(Number(char)) && char !== ' ') {
            currentNumber = currentNumber * 10 + (char.charCodeAt(0) - '0'.charCodeAt(0))
        } else if (char === '+') {
            result += sign * currentNumber
            currentNumber = 0
            sign = 1
        } else if (char === '-') {
            result += sign * currentNumber
            currentNumber = 0
            sign = -1
        } else if (char === '(') {
            stack.push(result)
            stack.push(sign)
            result = 0
            sign = 1
        } else if (char === ')') {
            result += sign * currentNumber
            currentNumber = 0
            result *= stack.pop()
            result += stack.pop()
        }
    }

    result += sign * currentNumber
    return result
};

File: 224-basic-calculator/basic-calculator.py
class Solution:
    def calculate(self, s: str) -> int:
        num = 0
        sign = 1
        res = 0
        stack = []

        for i in range(len(s)):
            c = s[i]

            if c.isdigit():
                num = num * 10 + int(c)
            elif c == '+':
                res += sign * num
                num = 0
                sign = 1
            elif c == '-':
                res += sign * num
                num = 0
                sign = -1
            elif c == '(':
                stack.append(res)
                stack.append(sign)
                res = 0
                sign = 1
            elif c == ')':
                res += sign * num
                num = 0
                prev_sign = stack.pop()
                prev_res = stack.pop()
                res = prev_res + prev_sign * res

        res += sign * num
        return res

            